nano/node/socket.cpp

```diff
@@ -109,8 +123,15 @@ void nano::socket::async_write (nano::shared_const_buffer const & buffer_a, std:
 				boost::asio::bind_executor (this_l->strand,
 				[buffer_a, cbk = std::move (callback), this_l] (boost::system::error_code ec, std::size_t size_a) {
 					--this_l->queue_size;
-					this_l->node.stats.add (nano::stat::type::traffic_tcp, nano::stat::dir::out, size_a);
-					this_l->stop_timer ();
+					if (ec)
+					{
+						this_l->node.stats.inc (nano::stat::type::tcp, nano::stat::detail::tcp_write_error, nano::stat::dir::in);
+					}
+					else
+					{
+						this_l->node.stats.add (nano::stat::type::traffic_tcp, nano::stat::dir::out, size_a);
+						this_l->stop_timer ();
```

Comment:
They are different kind of errors, it makes sense to increment both. One says there was a tcp write error, it doesn't mean that the socket closed. The caller then decides to to do nothing in case of tcp write error and let's the socket timeout. So I htink it makes sense to record both because they mean slightly different things and they are both correct.